src/app/components/lesson-card/LessonCardComponent.tsx

```diff
@@ -0,0 +1,114 @@
+import Hls from "hls.js";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { unlockedStatus } from "./constants";
+import { ILesson } from "../../services";
+import { Title, Content, Video } from "./styles";
+import { LockOutlined } from "@mui/icons-material";
+import { useDispatch } from "react-redux";
+import { coursePageSlice, useAppSelector } from "../../redux";
+import { Colors } from "../../common/assets";
+
+export const LessonCardComponent: React.FC<ILesson> = ({
+  id,
+  link,
+  order,
+  previewImageLink,
+  status,
+  title,
+}) => {
+  const [showVideo, setShowVideo] = useState(false);
+  const [downloadVideo, setDownloadVideo] = useState(false);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const { courseId } = useAppSelector(
+    (state) => state.persistedReducer.coursesListPage
+  );
+  const { coursesProgress } = useAppSelector(
+    (state) => state.persistedReducer.coursePage
+  );
+  const { setLessonProgress } = coursePageSlice.actions;
+
+  useEffect(() => {
+    if (showVideo) {
+      var video = document.getElementById(`video-${order}`);
+      if (Hls.isSupported() && video instanceof HTMLMediaElement) {
+        var hls = new Hls();
+
+        try {
+          hls.loadSource(link);
+          hls.attachMedia(video);
+          hls.on(Hls.Events.ERROR, function (event, data) {
+            if (data.type === Hls.ErrorTypes.NETWORK_ERROR) {
+              setShowErrorMessage(true);
+            }
+          });
+        } catch {
+          navigate("/error/");
+        }
+
+        video.currentTime = coursesProgress[courseId]
+          ? coursesProgress[courseId][id]
+          : 0;
+
+        video.addEventListener("timeupdate", function () {
+          dispatch(
+            setLessonProgress({
+              courseId,
+              lessonId: id,
+              lessonProgressTime: this.currentTime,
+            })
+          );
+        });
+      }
+    }
+  }, [downloadVideo]);
```

Comment:
To ensure that the component is updated correctly and avoid problems with dependencies, it is better to add all the variables that the useEffect function depends on to the dependency array

```suggestion
    }, [courseId, coursesProgress, dispatch, downloadVideo, id, link, navigate, order, setLessonProgress, showVideo]);
```